ctapipe/tools/dl1_merge.py

```diff
@@ -0,0 +1,359 @@
+            self.skip_file = True
+
+        # Gives warning if tables for listed nodes in 'optional_nodes'
+        # are missing but continues merging the rest of the file. If any
+        # other node from 'required_nodes' is missing, except the given combinations
+        # of traits and blocklist, the whole file will be skipped.
+        for node in required_nodes:
+            if self.is_simu is False and node in allowlist_simu:
+                continue
+            if self.skip_simu_images is True and node is blocklist_images[0]:
+                continue
+            if self.skip_images is True and node in blocklist_images:
+                continue
+            if self.skip_parameters is True and node in blocklist_parameters:
+                continue
+            if node in optional_nodes and node not in file:
+                self.log.warning(
+                    f"{node} is not in {file_path}. Continue with "
+                    "merging file. This table will be incomplete "
+                    "or empty"
+                )
+                continue
+            if node not in file:
+                self.log.warning(f"{node} is not in {file_path}. Skip file")
+                self.skip_file = True
+
+    def add_image_statistics(self, file):
+        # Creates table for image statistics and adds the entries together.
+        # This does not append rows to the existing table
+        image_statistics_path = allowlist_service[0]
+        if image_statistics_path in self.output_file:
+            table_out = self.output_file.root[image_statistics_path]
+            table_in = file.root[image_statistics_path]
+            for row in range(len(table_in)):
+                table_out.cols.counts[row] = np.add(
+                    table_out.cols.counts[row], table_in.cols.counts[row]
+                )
+                table_out.cols.cumulative_counts[row] = np.add(
+                    table_out.cols.cumulative_counts[row],
+                    table_in.cols.cumulative_counts[row],
+                )
+
+        elif "/dl1/service" not in self.output_file:
+            target_group = self.output_file.create_group(
+                "/dl1", "service", createparents=True
+            )
+            for node in allowlist_service:
+                file.copy_node(node, newparent=target_group)
+
+    def merge_tables(self, file):
+        # Loop over all nodes listed in required_nodes. Appends table to output_file
+        # if it already exists, otherwise creates group and copies node. If skip_images
+        # or skip_simu_images or skip_parameters flag is True, related groups
+        # will be skipped.
+        for node in required_nodes:
+            if node in file:
+                if self.skip_images is True and node in blocklist_images:
+                    continue
+
+                if self.skip_simu_images is True and node in blocklist_images[0]:
```

Comment:
Same as above, please don't rely on the order of things in these lists.

Reply:
Also, why are you using `node in  blocklist_images[0]`? that tests if a string is in another string, right?

Better use `startswith`, probably like this: `node.startswith('/dl1/image/tel/tel_'`)